crates/rpc/rpc/src/eth/cache.rs

```diff
@@ -380,6 +380,41 @@ where
     }
 }
 
+impl<Client, Tasks> EthStateCacheService<Client, Tasks>
+where
+    Client: StateProviderFactory + BlockProvider + EvmEnvProvider + Clone + Unpin + 'static,
+    Tasks: TaskSpawner + Clone + 'static,
+{
+    /// Check for new blocks and insert them in cache
+    async fn check_for_new_blocks<St>(&mut self, mut events: St)
+    where
+        St: Stream<Item = CanonStateNotification> + Unpin + 'static,
+    {
+        while let Some(event) = events.next().await {
+            match event {
+                CanonStateNotification::Commit { new } => {
+                    let (new_blocks, _) = new.inner();
```

Comment:
this has the entire block already so no need to fetch this from disk

what we need is new `CacheAction` commands to send the full block to the service